specfile/specfile.py

```diff
@@ -783,12 +783,16 @@ def update(value, requested_value, position):
                 and e.type == MacroDefinition
                 and not e.flip_pending
             }
-            regex, template, entities_to_flip = ValueParser.construct_regex(
-                value, modifiable_entities, flippable_entities, context=self
-            )
 
-            m = regex.match(requested_value)
-            if m:
+            # in case the value doesn't match after trying with flippable entities
+            # do a second pass without them
+            for flippable_ents in (flippable_entities, set()):
+                regex, template, entities_to_flip = ValueParser.construct_regex(
+                    value, modifiable_entities, flippable_ents, context=self
+                )
+                m = regex.match(requested_value)
+                if not m:
```

Comment:
I am wondering if walrus can help here..;)

Reply:
We can't use that, specfile still has to support Python 3.6.